BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/task_workdir_archive/task_workdir_archive.rb

```diff
@@ -52,4 +54,45 @@ def before_destroy_adjust_task #:nodoc:
     return true
   end
 
+  # compute archive file hash to avoid tampering
```

Comment:
While we're implementing a checksum method, move it to the SingleFile class so that all models can benefit.

Also:

* call it `cache_compute_md5sum` so we respect the conventions for method that accesses the cached content
* do NOT make it call `sync_to_cache()` it isn't this method's responsibility.
* let it raise an exception on any error, do not invoke 'addlog()', this would be poluting the log of the file
* remove the life cycle call back in task_workdir_archive (see other comment below)

Reply:
ok